flocker/acceptance/endtoend/test_leases.py

```diff
@@ -0,0 +1,119 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for the datasets REST API.
+"""
+
+# from uuid import UUID
+
+from twisted.trial.unittest import TestCase
+
+from ..testtools import (
+    require_cluster, require_moving_backend,  # create_dataset,
+    # REALISTIC_BLOCKDEVICE_SIZE,
+)
+
+
+class LeaseAPITests(TestCase):
+    """
+    Tests for the leases API.
+    """
+    @require_moving_backend
+    # require_cluster should probably delete leases as part of
+    # resetting the cluster state between tests
+    @require_cluster(2)
+    def test_lease_prevents_move(self, cluster):
+        """
+        A dataset cannot be moved if a lease is held on
+        it by a particular node.
+
+        Might look like this:
+
+        waiting_for_create = create_dataset(
+            self, cluster, maximum_size=REALISTIC_BLOCKDEVICE_SIZE)
+
+        def acquire_lease(dataset):
+            # Call the API to acquire a lease with the dataset ID.
+            # Give the lease an expiry of 60 seconds.
+            pass
+
+        def start_http_container(dataset):
+            # In the 60 seconds before the lease expires, we can launch
+            # the acceptance tests' data HTTP container and make POST requests
+            # to it in a looping call every second.
+            # return looping call deferred
+
+        def request_move_dataset(self):
+            # We can then request to move the dataset attached to the container.
+            return deferred
+
+        def wait_some_amount_of_time(self):
+            # Because the dataset is leased, we should be able to continue
+            # writing data via HTTP requests to the running container.
+            # We should be able to do this for some number of seconds.
+            # The looping call running in parallel to this is continuing to
+            # write data.
+
+            # wait some time and return a deferred
+
+        def stop_container(self):
+            # We stop the container, so that there is no constraint outside
+            # of leases to prevent the volume from being unmounted.
```

Comment:
i.e. Docker has bind mounted the `/flocker/dataset-id` directory which would prevent Flocker from unmounting the dataset, regardless of  a lease.